app/scripts/controllers/pod.js

```diff
@@ -86,7 +86,115 @@ angular.module('openshiftConsole')
         });
       }
     };
+       
+    $scope.onChange = function(arg) {     
+      // Iter. over all containers, setting new to active/connected & old's isActive to false
+      _.forEach($scope.containerTerminals, function(ctr){
+        if (ctr === arg) {
+          ctr.isActive = true;
+          
+          // If not previously connected to (Prevents duplicates)
+          if (!ctr.isConnected) {
+            ctr.isConnected = true;
+            // Add this to the list of connected terminals
+            $scope.connectedTerminals.push(ctr);
+          }
+        } else {
+          ctr.isActive = false;
+        }
+      }); 
+    }
+    
+    $scope.tabSelect = function() {
```

Comment:
I tested removing it and the terminal window will not change without it. It takes care of the user switching the terminal window to the one that they desire. The updateSelectTerminals handles refreshing the list of terminals available in the select on the event that the list of available terminals changes (Due to a terminal disconnect or container crash).